src/pykeen/evaluation/metrics.py

```diff
@@ -152,3 +167,207 @@ def get_ranking_metrics(ranks: np.ndarray) -> Mapping[str, float]:
     for metric_name, metric_func in ALL_TYPE_FUNCS.items():
         rv[metric_name] = metric_func(ranks).item()
     return rv
+
+
+class RankBasedMetric:
+    """A base class for rank-based metrics."""
+
+    #: whether it is increasing, i.e., larger values are better
+    increasing: ClassVar[bool] = False
+
+    #: the value range (as string)
+    value_range: ClassVar[Optional[ValueRange]] = None
+
+    #: the supported rank types. Most of the time equal to all rank types
+    supported_rank_types: ClassVar[Collection[RankType]] = RANK_TYPES
+
+    #: synonyms for this metric
+    synonyms: ClassVar[Collection[str]] = tuple()
+
+    #: whether the metric requires the number of candidates for each ranking task
+    needs_candidates: ClassVar[bool] = False
+
+    @abstractmethod
+    def __call__(self, ranks: np.ndarray, num_candidates: Optional[np.ndarray] = None) -> float:
+        """
+        Evaluate the metric.
+
+        :param ranks: shape: s
+            the individual ranks
+        :param num_candidates: shape: s
+            the number of candidates for each individual ranking task
+        """
+        raise NotImplementedError
+
+
+class ArithmeticMeanRank(RankBasedMetric):
+    """The (arithmetic) mean rank."""
+
+    value_range = ValueRange(lower=1, lower_inclusive=True, upper=math.inf)
+    synonyms = ("mean_rank", "mr")
+
+    @staticmethod
+    def call(ranks: np.ndarray) -> float:
+        """Evaluate the arithmetic mean rank."""
+        return np.mean(ranks).item()
+
+    def __call__(self, ranks: np.ndarray, num_candidates: Optional[np.ndarray] = None) -> float:  # noqa: D102
+        return ArithmeticMeanRank.call(ranks)
+
+
+class InverseArithmeticMeanRank(RankBasedMetric):
+    """The inverse arithmetic mean rank."""
```

Comment:
@cthoyt 

how would you proceed with the annotations? add them as docdata?

Reply:
can we just add more classvars?

Reply:
we can put the textual description in the docstring then parse it out :)

Reply:
2e96a36...34fde05

the textual description is often quite redundant to the name :smile:

Reply:
yeah its fine that the first line of the class docstring is the function description